app/src/main/kotlin/com/exactpro/th2/lwdataprovider/handlers/SearchMessagesHandler.kt

```diff
@@ -140,6 +140,7 @@ class SearchMessagesHandler(
                         val allLoaded = hashSetOf<Stream>()
                         do {
                             val continuePulling = pullUpdates(request, order, sink, allLoaded)
+                            Thread.sleep(configuration.keepOpenPullingTimeoutMs)
```

Comment:
note: please check if we should keep pulling before calling `sleep` method. As an alternative, I think you can call sleep method before calling `pullUpdates`. I think it would make sense to wait a bit before first pull attempt as well